// redux-test-utils/dist/utils/DispatchMock.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.createMockDispatch = void 0;
var fast_deep_equal_1 = __importDefault(require("fast-deep-equal"));
exports.createMockDispatch = function () {
    var actions = [];
    return {
        dispatch: function (action) {
            actions.push(action);
            return action;
        },
        getActions: function () {
            return actions;
        },
        getAction: function (type) {
            for (var i = 0; i < actions.length; i += 1) {
                if (actions[i].type === type) {
                    return actions[i];
                }
            }
            return undefined;
        },
        isActionTypeDispatched: function (type) {
            for (var i = 0; i < actions.length; i += 1) {
                if (actions[i].type === type) {
                    return true;
                }
            }
            return false;
        },
        isActionDispatched: function (action) {
            for (var i = 0; i < actions.length; i += 1) {
                if (actions[i].type === action.type) {
                    if (fast_deep_equal_1.default(actions[i], action)) {
                        return true;
                    }
                }
            }
            return false;
        },
    };
};
